src/test/scala/com/microsoft/hyperspace/index/IndexManagerTests.scala

```diff
@@ -478,6 +478,128 @@ class IndexManagerTests extends HyperspaceSuite with SQLHelper {
+      val df =
+        spark.read.option("spark.hyperspace.source.globbingPattern", globPath).parquet(p1)
+      val indexConfig = IndexConfig("index", Seq("RGUID"), Seq("imprs"))
+      val ex = intercept[HyperspaceException] {
+        hyperspace.createIndex(df, indexConfig)
+      }
+      assert(
+        ex.msg.contains("Some glob patterns do not match with available root paths of the " +
+          "source data."))
+    }
+  }
+
+  test("Verify createIndex works for globbing patterns with multiple paths.") {
+    withTempPathAsString { testPath =>
+      val absoluteTestPath = PathUtils.makeAbsolute(testPath)
+      val globPath1 = absoluteTestPath + "/1/*"
+      val globPath2 = absoluteTestPath + "/2/*"
+      val p1 = absoluteTestPath + "/1/1"
+      val p2 = absoluteTestPath + "/2/1"
+
+      // Store some data in p1 and p2 paths.
+      import spark.implicits._
+      SampleData.testData
+        .toDF("Date", "RGUID", "Query", "imprs", "clicks")
+        .limit(10)
+        .write
+        .parquet(p1)
+      SampleData.testData
+        .toDF("Date", "RGUID", "Query", "imprs", "clicks")
+        .limit(10)
+        .write
+        .parquet(p2)
+
+      // Create index with globbing pattern.
+      val df =
+        spark.read
+          .option("spark.hyperspace.source.globbingPattern", s"$globPath1,$globPath2")
+          .parquet(globPath1, globPath2)
+      val indexConfig = IndexConfig("index", Seq("RGUID"), Seq("imprs"))
+      hyperspace.createIndex(df, indexConfig)
+
+      // Check if latest log file contains data source as globbing pattern.
+      var index = latestIndexLogEntry(systemPath, indexConfig.indexName)
+      assert(index.relations.head.rootPaths.equals(Seq(globPath1, globPath2)))
+      assert(index.relations.head.data.properties.content.files.forall(path =>
+        path.toString.startsWith(p1) || path.toString.startsWith(p2)))
+
+      // Append data to a new directory which matches the globbing pattern.
+      val p3 = absoluteTestPath + "/1/3"
+      SampleData.testData
+        .toDF("Date", "RGUID", "Query", "imprs", "clicks")
+        .limit(3)
+        .write
+        .parquet(p3)
+
+      // Refresh index and check if files from both directories are present.
+      hyperspace.refreshIndex(indexConfig.indexName, "incremental")
+      index = latestIndexLogEntry(systemPath, indexConfig.indexName)
+      assert(index.relations.head.rootPaths.equals(Seq(globPath1, globPath2)))
+      assert(
+        index.relations.head.data.properties.content.files.forall(path =>
+          path.toString.startsWith(p1) || path.toString.startsWith(p2) || path.toString
+            .startsWith(p3)))
+      assert(index.relations.head.data.properties.content.files.exists(_.toString.startsWith(p1)))
+      assert(index.relations.head.data.properties.content.files.exists(_.toString.startsWith(p2)))
+      assert(index.relations.head.data.properties.content.files.exists(_.toString.startsWith(p3)))
+    }
+  }
+
```

Comment:
Should we validate hybrid scan scenario or no?

Reply:
for example, globbing pattern was `1/*`. Initially, you had `1/1`, and `1/2` was added. And validate if data under `1/2` was considered.

Reply:
If the query plan contains `1/1` and `1/2` properly, hybrid scan will work as expected.

Reply:
thanks @imback82 , @sezruby , from this discussion I will close this comment. Please reopen if additional test is required.

Reply:
yes, let's add one.

Reply:
added, thanks @imback82